hyper/http20/connection.py

```diff
@@ -522,6 +523,16 @@ def _consume_single_frame(self):
         # Parse the header. We can use the returned memoryview directly here.
         frame, length = Frame.parse_frame_header(header)
 
+        frame_size = self._settings[SettingsFrame.SETTINGS_MAX_FRAME_SIZE]
+        if (length > frame_size):
+            self._send_rst_frame(frame.stream_id, 6) # 6 = FRAME_SIZE_ERROR
+            log.warning(
+                "Frame size exceeded on stream %d (received: %d, max: %d)",
+                frame.stream_id,
+                length,
+                frame_size
+            )
```

Comment:
Oh, this check also isn't right. The value in `self._settings[]` is not a limit on the frame size the other side can *emit*, it's a limit on the size it can *receive*. `hyper` doesn't currently make the max frame size it can tolerate configurable (and I don't really feel like it should at this stage), so I think we just want to confirm that we tolerate MAX_FRAME_SIZE frames.

We then want to add a similar check to our own code for *sending*.